src/alire/alire-test_runner.adb

```diff
@@ -254,23 +274,33 @@ package body Alire.Test_Runner is
       -- Append --
       ------------
 
-      procedure Append (Dir_Entry : Adirs.Directory_Entry_Type) is
-         --  Helper function to append all .adb files in a folder
+      procedure Append (This         : Den.Walk.Item;
+                        Unused_Enter : in out Boolean;
+                        Unused_Stop  : in out Boolean)
+      is
+         --  Helper function to append all .adb files in a tree
          --  to the `Test_List` vector
 
-         Name : constant String := Adirs.Simple_Name (Dir_Entry);
+         Name : constant String :=
+                  Strip_Prefix (This.Path,
+                                Prefix => Root.Path / "src");
       begin
          if Name'Length > 4
            and then Name (Name'Last - 3 .. Name'Last) = ".adb"
            and then (Filter.Is_Empty
                      or else (for some F of Filter
                               => Ada.Strings.Fixed.Index (Name, F) /= 0))
          then
-            Test_List.Append (Name (Name'First .. Name'Last - 4));
+            Test_List.Append (Name (Name'First + 1 .. Name'Last - 4));
+            --  +1 removes initial '/'
```

Comment:
Why have a +1 here and no use `"src/"` in the call to `Strip_Prefix`?

Edit: Hmm there's the issue of windows file separators. Maybe it would be a lesser evil to do a `Replace ('\', '/')`, as Windows file APIs are mostly compatible with UNIX style path separators anyway

Reply:
That was to not do an extra `& Dir_Separator` elsewhere.

I don't think Windows should cause problems here? Or it would have triggered already something in the testsuite.

Reply:
Well, the testsuite did fail on windows because of path separators 😛 (not here though)

Reply:
Haha I saw it immediately after writing.

In the end I've changed it to always use forward slashes as it seems desirable that test output is consistent across OSes.